src/state.rs

```diff
@@ -456,6 +513,34 @@ impl DemandProxyState {
             .cloned()
     }
 
+    pub fn set_cached_resolve_dns_for_hostname(
+        &mut self,
+        hostname: String,
+        cached_atom_bool: SingleFlight,
+    ) {
+        let mut binding = self.state.write().unwrap();
+        let sf_map_element = binding
+            .resolved_dns
+            .crd_hashmap
+            .entry(hostname)
+            .or_insert(cached_atom_bool);
+        // increment 1 for every requests in concurrency
+        sf_map_element.calling_index += 1;
```

Comment:
Can we get rid of this index as we talked before? 

We may need to return the cached SingleFilight in this function

Reply:
Hi @hzxuzhonghu, according to our discussion before, we need to simplify the SingleFlight implementation, therefore, I removed the AtomicBool logic and keep the `Notify`. And as you know, all requests are coming in concurrency and are unorder, we finally have to know the specified request routine which really triggers the DNS resolving task, so the index is necessary.

Reply:
BTW, I also think to use channel similar with golang to implement this, but it's more complicated than directly use i32 index, so this should be the most simplified method.

Reply:
Understand this, and i can see the above op is not atomic, so a flag like this is needed